Examples/Mocks/AutoMockable.swift

```diff
@@ -0,0 +1,26 @@
+{% for type in types.based.AutoMockable %}{% if type.kind == 'protocol' %}
+{% if not type.name == "AutoMockable" %}
+class {{ type.name }}Mock: {{ type.name }} {
+
+    {% for variable in type.allVariables %}
+    var {{ variable.name }}: {{ variable.typeName }}
+    {% endfor %}
+
+    {% for method in type.allMethods %}
+    //MARK: - {{ method.shortName }}
+
+    {% if not method.shortName == "init" %}var {{ method.shortName }}Called = false{% endif %}
+    {%if method.parameters.count == 1 %}var {{ method.shortName }}Recieved{% for param in method.parameters %}{{ param.name|upperFirst }}: {{ param.typeName.unwrappedTypeName }}?{% endfor %}{% else %}{% if not method.parameters.count == 0 %}var {{ method.shortName }}RecievedArguments: ({% for param in method.parameters %}{{ param.name }}: {% if param.typeAttributes.escaping %}{{ param.unwrappedTypeName }}{% else %}{{ param.typeName }}{% endif %}{% if not forloop.last %}, {% endif %}{% endfor %})?{% endif %}{% endif %}
```

Comment:
I'm not sure `unwrappedTypeName` will work here as it will trim optional's stuff, what is not good for mocks, I guess, right?